Unable to link to postgres container with docker-engine 1.7.0 on Fedora 22

#### Description of problem:
I am using the recommended Fedora 22 docker-engine (docker-engine-1.7.0-1.fc22.x86_64.rpm) from the [Fedora installation page](http://docs.docker.com/installation/fedora/).

Following the instructions on the [Dockerizing Postgresql](http://docs.docker.com/examples/postgresql_service/) and [Linking Containers](https://docs.docker.com/userguide/dockerlinks/#communication-across-links) docs pages I am unable to connect to a postgres container from the simple bash container via the psql command. I received the error:

```
psql: could not connect to server: No route to host
        Is the server running on host "172.17.0.8" and accepting
        TCP/IP connections on port 5432?
```

#### docker version
```
Client version: 1.7.0
Client API version: 1.19
Go version (client): go1.4.2
Git commit (client): 0baf609
OS/Arch (client): linux/amd64
Server version: 1.7.0
Server API version: 1.19
Go version (server): go1.4.2
Git commit (server): 0baf609
OS/Arch (server): linux/amd64
```

#### docker info
```
Containers: 1
Images: 16
Storage Driver: devicemapper
 Pool Name: docker-253:1-1319522-pool
 Pool Blocksize: 65.54 kB
 Backing Filesystem: extfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 744.9 MB
 Data Space Total: 107.4 GB
 Data Space Available: 46.61 GB
 Metadata Space Used: 1.458 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.146 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.93 (2015-01-30)
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 4.0.6-300.fc22.x86_64
Operating System: Fedora 22 (Twenty Two)
CPUs: 4
Total Memory: 7.687 GiB
Name: *****
ID: NNLY:SH32:E4NY:4YV3:BQEJ:S5MV:QVOD:BQXZ:ED6T:H37F:4T5N:KABU
```

#### uname -a
```
Linux ***** 4.0.6-300.fc22.x86_64 #1 SMP Tue Jun 23 13:58:53 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

#### Environment details
Physical machine, Lenovo Thinkpad X220

#### How reproducible
Every time

#### Steps to Reproduce
1. Install docker-engine 1.7.0 as per the instructions for [Fedora 22 on installation page](http://docs.docker.com/installation/fedora/)
2. Follow the instructions on the [Dockerizing PostgreSQL page](http://docs.docker.com/examples/postgresql_service/)

#### Actual Results
Unable to connect to postgres container:
```
psql: could not connect to server: No route to host
        Is the server running on host "172.17.0.8" and accepting
        TCP/IP connections on port 5432?
```

#### Expected Results
Should connect to the postgres container and display the psql prompt.

#### Additional info
Tested on Mac OSX (Yosemite) with boot2docker 1.7.0 and Fedora 22 docker 1.6.0 from the fedora repo (installed using `sudo dnf install docker`) and in both cases it works fine.

@mrjana could you link to instructions? I tried to build from source with a simple ```make``` at source root, it failed at step 42
```
Step 42 : COPY . /go/src/github.com/docker/docker
 ---> 762b505e839b
Removing intermediate container 415d5421fc37
Successfully built 762b505e839b
docker run --rm -it --privileged -e BUILDFLAGS -e DOCKER_CLIENTONLY -e DOCKER_DEBUG -e DOCKER_EXECDRIVER -e DOCKER_EXPERIMENTAL -e DOCKER_GRAPHDRIVER -e DOCKER_STORAGE_OPTS -e DOCKER_USERLANDPROXY -e TESTDIRS -e TESTFLAGS -e TIMEOUT -v "/root/docker/bundles:/go/src/github.com/docker/docker/bundles" "docker-dev:master" hack/make.sh binary

---> Making bundle: binary (in bundles/1.8.0-dev/binary)
Building: bundles/1.8.0-dev/binary/docker-1.8.0-dev
go build github.com/docker/docker/docker: /usr/local/go/pkg/tool/linux_amd64/6l: signal: killed
Makefile:44: recipe for target 'binary' failed
make: *** [binary] Error 1
```

Maybe it will be easier that you reproduce with the test instructions on a Fedora 22?

--- a/hack/vendor.sh
+++ b/hack/vendor.sh
@@ -21,7 +21,7 @@ clone git golang.org/x/net 3cffabab72adf04f8e3b01c5baf775361837b5fe https://gith
 clone hg code.google.com/p/gosqlite 74691fb6f837
 
 #get libnetwork packages
-clone git github.com/docker/libnetwork 2a5cb84758b5115d99d8f82c84845417c6c345a3
+clone git github.com/docker/libnetwork f1c5671f1ee2133055144e566cd8b3a0ae4f0433
 clone git github.com/armon/go-metrics eb0af217e5e9747e41dd5303755356b62d28e3ec
 clone git github.com/hashicorp/go-msgpack 71c2886f5a673a35f909803f38ece5810165097b
 clone git github.com/hashicorp/memberlist 9a1e242e454d2443df330bdd51a436d5a9058fc4

